src/software/SfM/main_ComputeFeatures.cpp

```diff
@@ -56,6 +60,9 @@ int main(int argc, char **argv)
   std::string sImage_Describer_Method = "SIFT";
   bool bForce = false;
   std::string sFeaturePreset = "";
+#ifdef OPENMVG_USE_OPENMP
+  int iNumThreads = 0;
```

Comment:
I was using `iNumThreads  = 0` to disable parallel block entirely on line 237.  [omp_in_parallel()](https://msdn.microsoft.com/en-us/library/y22bxh22.aspx) would return false in "parallelized" for loop, and would behave exactly the same as non-parallelized version. Also nested for loop with existing parallelization should run in parallel, which they would normally not if `omp_in_parallel` returns true/nonzero - except if explicitly set with [omp_set_nested(int)](https://msdn.microsoft.com/en-us/library/sk3zt8e1.aspx). But yeah, I did not check whether `omp_set_num_threads(int)` can receive 0 as input. Its probably best to set default to 1.